nodejs/awsx/apigateway/README.md

```diff
@@ -0,0 +1,139 @@
+            return {
+                statusCode: 200,
+                body: "hello",
+            };
+        },
+    }],
+})
+```
+
+A complete example can be found [here](https://github.com/pulumi/pulumi-awsx/blob/master/nodejs/awsx/examples/api/index.ts).
+
+#### Static Route
+
+A Static Route is a route that will map to static content in files/directories. You will need to define the local path & then the files (and subdirectories) will be uploaded into S3 objects. If the local path points to a file, you can specify the content-type. Else, the content types for all files in a directory are inferred.
+
+By default, any request on directory will serve index.html. This behavior can be disabled, by setting the index field to false.
+
+```ts
+import * as aws from "@pulumi/aws";
+import * as awsx from "@pulumi/awsx";
+
+let endpoint = new awsx.apigateway.API("example", {
+    routes: [{
+        path: "/www",
+        localPath: "www",
+        index: false,
+    }],
+})
+```
+
+A complete example can be found [here](https://github.com/pulumi/pulumi-awsx/blob/master/nodejs/awsx/examples/api/index.ts).
+
+#### Integration Route
+
+[TODO]
+
+#### Raw Data Route
+
+[TODO]
+
+### Request Validation
+
+API Gateway can perform basic validations against request parameters, a request payload or both. When a validation fails, a 400 error is returned immediately.
+
+#### Validators
+
+Validators can be assigned at the API level or at the method level. The validators defined at a method level override any validator set at the API level.
+
+You must specify one of the following validators:
+
+* "ALL" - validate both the request body and request parameters
+* "BODY_ONLY" - validate only the request body
+* "PARAMS_ONLY" - validate only the request parameters
+
+```ts
+import * as aws from "@pulumi/aws";
+import * as awsx from "@pulumi/awsx";
+
+let endpoint = new awsx.apigateway.API("example", {
+    routes: [{
+        path: "/www",
+        localPath: "www",
+        index: false,
+        requestValidator: "PARAMS_ONLY",
+    }],
+    requestValidator: "ALL",
+})
+```
+
+#### Request Parameters
+
+For each required request parameter, you must define the name and where the parameter is expected (i.e. "path", "query", or "header").
+
+```ts
+import * as aws from "@pulumi/aws";
+import * as awsx from "@pulumi/awsx";
+
+let endpoint = new awsx.apigateway.API("example", {
+    routes: [{
+        path: "/www",
+        localPath: "www",
+        index: false,
+        requestValidator: "PARAMS_ONLY",
+        requiredParams: [{
+            name: "key",
+            in: "query",
+        }]
+    }],
+})
+```
+
+#### Request Body
+
+[TODO]
+
+### Swagger String
+
+You can use a OpenAPI specification that is in string form to initialize the API Gateway. This in a way is an escape hatch for implementing featured not yet supported by Pulumi. You must manually provide permission for any route targets to be invoked by API Gateway when using this option.
+
```

Comment:
this part is fine.  And i like that it would be at the end.  I wouldn't even mention anything about swagger earlier on.